identity-core/src/crypto/merkle_key/signer.rs

```diff
@@ -121,13 +122,19 @@ where
   K: MerkleSigningKey<D, S>,
   S::Output: AsRef<[u8]>,
 {
-  fn sign<X>(data: &X, private: &K) -> Result<SignatureValue>
+  type SignError = SigningError;
+  type SignatureCreationError = SigningError;
```

Comment:
Is the `SignatureCreationError` intended for forwards compatibility? Here it's the same type as `SignError`, and I'm not sure if signing and creating a signature are different enough to warrant the different error types.

Reply:
Changed the trait to use `SigningError` as its error type.